MoreCyclopsUpgrades/Config/ModConfig.cs

```diff
@@ -0,0 +1,239 @@
+﻿namespace MoreCyclopsUpgrades.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using Common;
+    using MoreCyclopsUpgrades.Config.ChoiceEnums;
+    using CommonCyclopsUpgrades.Options;
+    using UnityEngine;
+    using CommonCyclopsUpgrades;
+
+    internal class ModConfig : IModConfig
+    {
+        private static readonly ModConfig main = new ModConfig();
+        internal static IModConfig Main
+        {
+            get
+            {
+                if (!main.initialized)
+                    main.Initialize();
+
+                return main;
+            }
+        }
+
+        private bool initialized = false;
+
+        private const string AuxConsoleEnabledKey = "AuxConsoleEnabled";
+        private const string ChallengeModeKey = "ChallengeMode";
+        private const string DeficitThresholdKey = "DeficitThreshold";
+        private const string ChargerIconsKey = "ShowChargerIcons";
+        private const string DebugLogsEnabledKey = "EnableDebugLogs";
+        private const string HelmEnergyDisplayKey = "HelmEnergyDisplay";
+
+        private readonly ToggleOption auxConsoleEnabled = new ToggleOption(AuxConsoleEnabledKey, "Enable Aux Upgrade Console (Requires restart)")
+        {
+            State = true
+        };
+        private readonly ChoiceOption challengeMode = new ChoiceOption(ChallengeModeKey, "Challenge Level (Requires restart)")
```

Comment:
Under the current set up, this could be updated to not require a full restart by dynamically updating all ChargerManagers with the new value.  
It would make testing easier too.